### PR TITLE
generic gateway as a separate component

### PR DESCRIPTION
**Pull request by dgkanatsios**
charts for Generic Gateway, as per 
https://github.com/Azure/brigade/pull/762#issuecomment-454799925
https://github.com/Azure/brigade/issues/602

this should be merged when https://github.com/Azure/brigade/pull/762 is merged

### CODE REVIEW

File: charts/brigade/templates/gateway-generic-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{{ if .Values.genericGateway.enabled }}
+{{ $fullname := include "brigade.genericGateway.fullname" . }}
```
Comment (vdice) [inline charts/brigade/templates/gateway-generic-deployment.yaml:2]: We need to define `"brigade.genericGateway.fullname"` in `_helpers.tpl`, perhaps like so:
```
{{- define "brigade.genericGateway.fullname" -}}
{{ include "brigade.fullname" . | printf "%s-generic-gateway" }}
{{- end -}}
```

**Review by vdice — COMMENTED** [review]
Looking good!

I'd request we add the `genericWebhookSecret` entry to the `brigade-project` chart (secret.yaml):
```
  {{if .Values.genericWebhookSecret -}}
  genericWebhookSecret: {{ .Values.genericWebhookSecret | quote }}
  {{- end }} 
```
(in addition to a corresponding entry in the chart's `values.yaml` file.)

Other than that, I'd say both revisions to their respective charts warrant a minor version bump.  The charts can be built and this repo re-indexed via `make build index` (Yes, go ahead and commit other renewed chart artifacts as well...)

File: charts/brigade/templates/gateway-generic-deployment.yaml (hunk 1 above)
Comment (dgkanatsios) [reply charts/brigade/templates/gateway-generic-deployment.yaml:2]: thanks, I was wondering where I should place that... May I ask what method did you use to find it (other than `helm install charts/brigade --dry-run`)?

Comment (vdice) [reply charts/brigade/templates/gateway-generic-deployment.yaml:2]: @dgkanatsios I "for reals" installed for testing these PRs, but yes, `--dry-run` will catch it _if_ this feature is enabled when running the command: 
```
 $ helm install -n testbrigade charts/brigade --set genericGateway.enabled=true --dry-run
Error: render error in "brigade/templates/gateway-generic-service.yaml": template: brigade/templates/gateway-generic-service.yaml:5:20: executing "brigade/templates/gateway-generic-service.yaml" at <{{template "brigade....>: template "brigade.genericGateway.fullname" not defined
```
We have a ticket here to add a test target running `--dry-run` installs to catch these things -- but this is a tricky case of a certain flag is needed as well... maybe for ci we turn 'em all on?

Comment (dgkanatsios) [reply charts/brigade/templates/gateway-generic-deployment.yaml:2]: got it thanks. Had some difficulty in running --dry-run though, mentioned about Kashti missing (although it's of course there in the charts/ folder), will try again

File: charts/brigade-project/templates/secret.yaml [hunk 2]
```diff
@@ -65,3 +65,6 @@ stringData:
   {{if .Values.workerCommand -}}
   workerCommand: {{.Values.workerCommand | quote }}
   {{- end }}
+  {{if .Values.genericGatewaySecret -}}
```
Comment (vdice) [inline charts/brigade-project/templates/secret.yaml:68]: Let's add an entry for `genericGatewaySecret` to `charts/brigade-project/values.yaml` as well...

File: charts/brigade-project/values.yaml [hunk 3]
```diff
@@ -27,6 +27,10 @@ cloneURL: "https://github.com/deis/empty-testbed.git"
 # Used by GitHub and other services to compute hook HMACs.
 sharedSecret: "IBrakeForSeaBeasts"
 
+# OPTIONAL: genericGatewaySecret is the secret that should be used on subsequent calls to the Generic Gateway in order to raise a "webhook" event for this Project
+# This is REQUIRED for the Generic Gateway, but optional otherwise.
+genericGatewaySecret: "mygenericsecret"
```
Comment (vdice) [inline charts/brigade-project/values.yaml:32]: perfect!

**Review by vdice — APPROVED** [review]
Lastly, @dgkanatsios would you increment both brigade & brigade-project chart minor versions, run `make build index` and add a commit with the changes?  Otherwise, LGTM!

**Review by vdice — APPROVED** [review]

**Comment by vdice** [comment]
Oops, almost forgot about this PR after the feature had been merged in Brigade.  Merging now -- thanks again @dgkanatsios !

**Pull request merged**